src/units/unit.cpp

```diff
@@ -994,22 +1019,12 @@ void unit::advance_to(const unit_type& u_type, bool use_traits)
 
 	// Now that modifications are done modifying traits, check if poison should
 	// be cleared.
-	if(get_state("unpoisonable")) {
-		set_state(STATE_POISONED, false);
-	}
-	if(get_state("unslowable")) {
-		set_state(STATE_SLOWED, false);
-	}
+	// Make sure apply_modifications() didn't attempt to heal the unit (for example if the unit has a default amla.).
+	ss();
 	if(get_state("unpetrifiable")) {
 		set_state(STATE_PETRIFIED, false);
```

Comment:
Why is petrification treated differently than poison and slow?

Reply:
I just wasn't sure about petrification since i don't know how its usually used, with slow and poisoned i can easily say "it usually applied via attacks and not via objects", for petrification i cant. (I know that there exists a petrifies weapon special, bu i just never have seen it and by default is just compltely unbalances/gamebeaking)

Reply:
I think it makes sense to treat petrify and slow the same. And in 1.19 change default petrifying to limited turn duration.

Reply:
We can't just change petrification to limited duration. Several multiplayer scenarios rely on it being permanent.